src/key/key.js

```diff
@@ -360,47 +359,77 @@ class Key {
     }
     // check for valid, unrevoked, unexpired self signature
     const { selfCertification } = await this.getPrimaryUser(date, userID, config);
-    // check for expiration time
+    // check for expiration time in binding signatures
     if (helper.isDataExpired(primaryKey, selfCertification, date)) {
       throw new Error('Primary key is expired');
     }
+    // check for expiration time in direct signatures
+    if (this.directSignatures.length > 0) {
+      const directSignature = await helper.getLatestValidSignature(
+        this.directSignatures, primaryKey, enums.signature.key, { key: primaryKey }, date, config
+      ).catch(() => {}); // invalid signatures are discarded, to avoid breaking the key
+
+      if (directSignature && helper.isDataExpired(primaryKey, directSignature, date)) {
+        throw new Error('Primary key is expired through a direct signature');
```

Comment:
This is a bit nitpicky, but this error message seems slightly too specific to me. I don't think the user really cares what kind of signature was used to make the key expire